bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/types/BRecordType.java

```diff
@@ -76,10 +76,10 @@ public BRecordType(String typeName, Module pkg, long flags, boolean sealed, int
     public BRecordType(String typeName, Module pkg, long flags, Map<String, Field> fields, Type restFieldType,
```

Comment:
Same applies for object types too.

Reply:
we plan to address this issue separately based on this issue https://github.com/ballerina-platform/ballerina-lang/issues/27029
This is because there are issue with setting fields on a record when the record is set to be `readonly`

Reply:
I changed for the `BRecordType`. As for the object type we don't initialize the fields via the constructor.